process/caller.py

```diff
@@ -158,17 +176,36 @@ def call_models_and_write_output(self, xc: np.ndarray, ifail: int) -> None:
                     finalise(self.models, ifail)
                     return
 
-                # Mfiles not yet idempotent: re-evaluate models
+                # Mfiles not yet idempotent: need to re-evaluate models
                 logger.debug("Mfiles not idempotent, evaluating models again")
-                previous_mfile_arr = np.copy(current_mfile_arr)
+                previous_mfile_data = mfile_data.copy()
 
-            raise RuntimeError(
+            # Values haven't all stabilised after 10 evaluations
+            # Which variables are still changing?
+            non_idempotent_warning = (
                 "Model evaluations at the current optimisation parameter vector "
                 "don't produce idempotent values in the final output."
             )
+            non_idempotent_table = tabulate(
+                [[k, v[0], v[1]] for k, v in nonconverged_vars.items()],
+                headers=["Variable", "Previous value", "Current value"],
+            )
+
+            warnings.warn("\033[93m" + non_idempotent_warning)
+            print(non_idempotent_table + "\033[0m")
```

Comment:
Why is all of the output not handled by the `warn` function?

Reply:
Quite right, I've combined these.